external/upstream/integratorxx/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+
+if(${ENABLE_IntegratorXX})
+    find_package(IntegratorXX CONFIG)
+
+    if(${IntegratorXX_FOUND})
+        get_property(_loc TARGET IntegratorXX::IntegratorXX PROPERTY LOCATION)
+        message(STATUS "${Cyan}Found Integratxx${ColourReset}: ${_loc} (found version ${IntegratorXX_VERSION})")
+        add_library(integratorxx_external INTERFACE)  # dummy
+    else()
+
+        if(${CMAKE_INSIST_FIND_PACKAGE_IntegratorXX})
+            message(FATAL_ERROR "Suitable IntegratorXX could not be externally located as user insists")
+        endif()
+    
+        include(ExternalProject)
+        message(STATUS "Suitable IntegratorXX could not be located, ${Magenta}Building IntegratorXX${ColourReset} instead.")
+        ExternalProject_Add(integratorxx_external
+            URL https://github.com/wavefunction91/IntegratorXX/archive/master.tar.gz
+            UPDATE_COMMAND ""
+            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
+                       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                       -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                       -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
+                       -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
+                       -DCMAKE_CXX_STANDARD=${psi4_CXX_STANDARD}
```

Comment:
@loriab FWIW - I don't respect this flag in `IntegratorXX` - it's a C++17 library, you could conceivably compile with 20 / 2b, but definitely not with 14. Would you prefer that I have a check on my end if you're requesting  < 17 and bail? Or should `ENABLE_IntegratorXX` be contingent on 17+?